pkg/hubble/cell/hubble.go

```diff
@@ -0,0 +1,619 @@
+	var metricsTLSConfig *certloader.WatchedServerConfig
+	if h.config.EnableMetricsServerTLS {
+		metricsTLSConfigChan, err := certloader.FutureWatchedServerConfig(
+			h.log.WithField("config", "hubble-metrics-server-tls"),
+			h.config.MetricsServerTLSClientCAFiles,
+			h.config.MetricsServerTLSCertFile,
+			h.config.MetricsServerTLSKeyFile,
+		)
+		if err != nil {
+			h.log.WithError(err).Error("Failed to initialize Hubble metrics server TLS configuration")
+			return
+		}
+		waitingMsgTimeout := time.After(30 * time.Second)
+		for metricsTLSConfig == nil {
+			select {
+			case metricsTLSConfig = <-metricsTLSConfigChan:
+			case <-waitingMsgTimeout:
+				h.log.Info("Waiting for Hubble metrics server TLS certificate and key files to be created")
+			case <-ctx.Done():
+				h.log.WithError(ctx.Err()).Error("Timeout while waiting for Hubble metrics server TLS certificate and key files to be created")
+				return
+			}
+		}
+		go func() {
+			<-ctx.Done()
+			metricsTLSConfig.Stop()
+		}()
+	}
+
+	var srv *http.Server
+	if h.config.MetricsServer != "" {
+		h.log.WithFields(logrus.Fields{
+			"address": h.config.MetricsServer,
+			"metrics": h.config.Metrics,
+			"tls":     h.config.EnableMetricsServerTLS,
+		}).Info("Starting Hubble Metrics server")
+
+		err := metrics.InitMetrics(metrics.Registry, api.ParseStaticMetricsConfig(h.config.Metrics), grpcMetrics)
+		if err != nil {
+			h.log.WithError(err).Error("Unable to setup metrics: %w", err)
+			return
+		}
+
+		srv = &http.Server{
+			Addr:    h.config.MetricsServer,
+			Handler: nil,
+		}
+		metrics.InitMetricsServerHandler(srv, metrics.Registry, h.config.EnableOpenMetrics)
+
+		go func() {
+			if err := metrics.StartMetricsServer(srv, h.log, metricsTLSConfig, grpcMetrics); err != nil && !errors.Is(err, http.ErrServerClosed) {
+				h.log.WithError(err).Error("Hubble metrics server encountered an error")
+				return
+			}
+		}()
+
+		observerOpts = append(observerOpts,
+			observeroption.WithOnDecodedFlowFunc(func(ctx context.Context, flow *flowpb.Flow) (bool, error) {
+				err := metrics.ProcessFlow(ctx, flow)
+				if err != nil {
+					h.log.WithError(err).Error("Failed to ProcessFlow in metrics handler")
+				}
+				return false, nil
+			}),
+		)
+
+		localSrvOpts = append(localSrvOpts,
+			serveroption.WithGRPCMetrics(grpcMetrics),
+			serveroption.WithGRPCStreamInterceptor(grpcMetrics.StreamServerInterceptor()),
+			serveroption.WithGRPCUnaryInterceptor(grpcMetrics.UnaryServerInterceptor()),
+		)
+	}
+
+	if h.config.EnableRedact {
+		parserOpts = append(
+			parserOpts,
+			parserOptions.Redact(
+				h.log,
+				h.config.RedactHttpURLQuery,
+				h.config.RedactHttpUserInfo,
+				h.config.RedactKafkaAPIKey,
+				h.config.RedactHttpHeadersAllow,
+				h.config.RedactHttpHeadersDeny,
+			),
+		)
+	}
+
+	payloadParser, err := parser.New(h.log, h, h, h, h.ipcache, h, link.NewLinkCache(), h.cgroupManager, h.config.SkipUnknownCGroupIDs, parserOpts...)
```

Comment:
From `d, d, d, d, ...` to `h, h, h, h, ...`... I'm sure @glibsm will appreciate 🤣

Reply:
probably would have been already possible to move more of the functions that have been moved from `Daemon` -> `Hubble` further down to the `Hubble Parser` and let the parser depend on these components instead of the fine-grained interfaces (reducing the amount of arguments)

But I think it's up to your team to decide the way forward for the more fine-grained modularization within the Hubble Cell :+1: I am very confident that you will find a good solution :rocket: